ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java

```diff
@@ -496,6 +498,21 @@ public void shouldGenerateCorrectCodeForDecimalUnaryPlus() {
     assertThat(java, is("(TEST1_COL8.plus(new MathContext(2, RoundingMode.UNNECESSARY)))"));
   }
 
+  @Test
+  public void shouldGenerateCorrectCodeForDecimalCast() {
```

Comment:
how about a negative tests that test the two if statements you've added/changed in the class?

Also, is it worth having a test to ensure CAST  from decimal to a decimal with the same scale and precision is a no op?

(Easier if you pull out the logic into a util and add the tests there...)

Reply:
the last bit can be tested in `DecimalUtilTest` - I have added a test for that

Reply:
The former is impossible - we will never have a SqlType that resolves to `DECIMAL` and whose type is not decimal (so impossible I can't test it without mocking a bizzaro type). I think that's a little bit of test bloat.

Reply:
For the former... if its impossible, why is there an if block for it? :D

Reply:
so IntelliJ doesn't complain about unchecked casts :P